Reference/google/protobuf/any_test.pb.swift

```diff
@@ -123,35 +108,42 @@ struct ProtobufUnittest_TestAny: SwiftProtobuf.Proto3Message, SwiftProtobuf._Mes
   init() {}
 
   mutating func _protoc_generated_decodeMessage<D: SwiftProtobuf.Decoder>(decoder: inout D) throws {
-    try _uniqueStorage().decodeMessage(decoder: &decoder)
+    try withExtendedLifetime(_storage) { (_storage: _StorageClass) in
+      while let fieldNumber = try decoder.nextFieldNumber() {
+        try decodeField(decoder: &decoder, fieldNumber: fieldNumber)
+      }
+    }
   }
 
   mutating func _protoc_generated_decodeField<D: SwiftProtobuf.Decoder>(decoder: inout D, fieldNumber: Int) throws {
-    try _uniqueStorage().decodeField(decoder: &decoder, fieldNumber: fieldNumber)
+    switch fieldNumber {
```

Comment:
If we were to go that route, I think we may as well just write a `decodeGroup` method instead that does the same thing as `decodeMessage` but with a different terminating condition (look for the right end-tag). The amount of increased codegen would be about the same and that would make a little more sense to the reader.

I think we would still need `withExtendedLifetime`; I'm not sure the change you suggested would affect the way reference counts are added/subbed by the parameter passing.